design/one-pager-automatic-dependency-downgrades.md

```diff
@@ -0,0 +1,83 @@
+address this issue by introducing automatic dependency version downgrades as a configuration option.
+
+<img src="images/package-manager.png" alt="Package Manager diagram" width="650" />
+
+## Possible Issues
+
+During dependency version downgrades, the following issues may occur:
+
+1) Missing CRDs
+2) Missing CRD fields
+3) CRD storage version changes
+
+Please note that these issues may also occur during package upgrades. But in that case, we assume the user is aware of 
+the changes and accepts the risks.
+
+Detecting these issues in the package manager before downgrading dependencies would significantly complicate the package 
+manager's logic and increase the implementation complexity. It would also require downloading the package base layer 
+multiple times with different versions, leading to slower performance. Therefore, we suggest not implementing pre-checks 
+for these issues before downgrading dependencies. Instead, we will document the potential issues and risks, providing 
+users with the necessary information to make informed decisions.
+
+### Missing CRDs
+
+The CRDs installed with the newer version may not exist in the older one. This can result in custom resources 
+without associated controllers, potentially causing unexpected behavior, such as lack of updates on the cloud 
+provider's side after updating the MR. Updating the parent package to upgrade its dependencies to the newer version
+resolves this issue.
+
+### Missing CRD Fields
+
+CRD fields may change between versions. If the newer version introduces additional fields, the older version will remove
+them. Downgrading the package can lead to the loss of these values, and upgrading back to the newer version 
+cannot restore them to their previously assigned states.
+ 
+### CRD Storage Version Changes
+
+CRD storage versions may change between versions. If the storage version is upgraded in the newer version, trying to 
+downgrade the package will not automatically change the storage version on Kubernetes and fail. Some providers, such as
+the ones in the [Upbound Official program](https://marketplace.upbound.io/providers?tier=official), prevent this issue 
+by ensuring the storage version is not changed between minor versions. However, this is not guaranteed for all Crossplane
+providers.
+
+## Proposal
+
+The automatic dependency version downgrade feature will be implemented as a configuration option. The user can enable this
+configuration via a helm to set the `AutomaticDependencyDowngrade` environment variable as `true` in the package manager
+configuration. This functionality will be disabled by default and needs to be enabled together with the 
+`--enable-dependency-version-upgrades` feature flag. Enabling the configuration means that users accept the risks 
+associated with automatic dependency version downgrades.
+
+When the configuration is enabled, the package manager will automatically downgrade the dependencies to the maximum valid 
```

Comment:
could you give a tangible example of a scenario that would cause this to happen, and what the result would be? e.g. package A has dependency X, then...